File: APLOS/Areas/Machines/Scripts/machineMasterControllers.js
'use strict';
machineMasterControllers.$inject = ['cboService', 'commonMessage', '$scope', '$rootScope', 'baseService', '$routeParams', '$location', '$http', '$filter'];
function machineMasterControllers(cboService, commonMessage, $scope, $rootScope, baseService, $routeParams, $location, $http, $filter) {
    $rootScope.title = 'Machine Master'; 
    $scope.Action = 'Save';  
    $scope.Action1 = 'Save';
    $scope.OperationActivityList = [];
    $scope.OperationTypeList = [];
    $scope.OperationCategoryList = [];
    $scope.SkillList = [];
    $scope.MachineMasterList = [];
    $scope.ProcessList = [];
    $scope.legalDesignationList = [];
    $scope.SkillGroupingList = [];
    $scope.GetDataByMasterOrderIdList = [];
    $scope.EntityList = [];
    $scope.PositionList = [];
   
  
    $scope.path = 'Machines/MachineMaster/';//ControlerName
    $scope.saveUrl = $scope.path + 'Create';
    $scope.updateUrl = $scope.path + 'Edit';
    $scope.deleteUrl = $scope.path + 'Delete/';
    $scope.saveUrl1 = $scope.path + 'CreateManpower';
    $scope.updateUrl1 = $scope.path + 'EditManpower';
    $scope.deleteUrl1 = $scope.path + 'DeleteManpower/';
    $scope.model = {
        Id: null,
        CompanyGroupId: null,
        Sequence: null,
        Code: null,
        ShortName: null,
        StandardName: null,
        UserName: null,
        OperationActivityId: null,
        OperationTypeId: null,
        OperationCategoryId: null,
        SkillId: null,
        Type: null,
        MachineMasterId: null,
        SkillGroupId: null,
        LegalDesignationId: null,
        ProcessId: null,
        ProposedSalary: null,
        Remarks: null,
        Active: null
    };
    $scope.modelNew = Object.assign({}, $scope.model);

    $scope.modelM = {
        Id: null,       
        CompanyGroupId:null,
        Sequence: null,
        OperationMasterId: null,
      EntityId: null,
      PositionId: null,
     Caption: null,
      ManpowerBudget: null,
      Active: null      
    };
    $scope.modelNewM = Object.assign({}, $scope.modelM);
    
   

 
  
    // #region GET Display DTA ON GRID
    $scope.GriddataOperationMaster = [];
    $scope.getaldataOperationMaster = function () {
        debugger;
        $http({
            method: "GET",
            dataType: 'JSON',
            //url: $scope.getSearchListUrl,
            url: 'IE/OperationMaster/GetOperationMaster', 
        }).then(function successCallback(response) {
            $scope.GriddataOperationMaster = response.data;
           
            //entrydata = copy(searchdata);
        });
    };
    $scope.getaldataOperationMaster();

    $scope.GetOperationPositionMp = [];
    $scope.GetOperationPositionMPBudget = function () {
        $http({
            method: "GET",
            dataType: 'JSON',
            //url: $scope.getSearchListUrl,
            url: 'IE/OperationMaster/GetOperationPositionMPBudget',
        }).then(function successCallback(response) {
            $scope.GetOperationPositionMp = response.data;
            //entrydata = copy(searchdata);
        });
    };
    $scope.GetOperationPositionMPBudget();

    
//#endregion



    // #region Bind Data on DropdownList 

    $scope.EntityCbo = function () {
        $http({
            method: 'GET',
            url: 'IE/OperationMaster/GetCboEntity'
        }).then(function successCallback(response) {
            $scope.EntityList = response.data;
        });
    }
    $scope.EntityCbo();
    $scope.PositionCbo = function () {
        $http({
            method: 'GET',
            url: 'IE/OperationMaster/GetCboPosition'
        }).then(function successCallback(response) {
            $scope.PositionList = response.data;
        });
    }
    $scope.PositionCbo();
    
    $scope.OperationActivityCbo = function () {
        $http({
            method: 'GET',
            url: 'IE/OperationMaster/GetCboOperationActivity'
        }).then(function successCallback(response) {
            $scope.OperationActivityList = response.data;
        });
    }
    $scope.OperationActivityCbo();

    $scope.GetCboOperationTypeCbo = function () {
        $http({
            method: 'GET',
            url: 'IE/OperationMaster/GetCboOperationType'
        }).then(function successCallback(response) {
            $scope.OperationTypeList = response.data;
        });
    }
    $scope.GetCboOperationTypeCbo();



    $scope.GetCboOperationCategoryCbo = function () {
        $http({
            method: 'GET',
            url: 'IE/OperationMaster/GetCboOperationCategory'
        }).then(function successCallback(response) {
            $scope.OperationCategoryList = response.data;
        });
    }
    $scope.GetCboOperationCategoryCbo();


    $scope.GetCboSkillCbo = function () {
        $http({
            method: 'GET',
            url: 'IE/OperationMaster/GetCboSkill'
        }).then(function successCallback(response) {
            $scope.SkillList = response.data;
        });
    }
    $scope.GetCboSkillCbo();



    $scope.GetCboMachineMasterCbo = function () {
        debugger;
        $http({
            method: 'GET',
            url: 'IE/OperationMaster/GetCboMachineMaster'
        }).then(function successCallback(response) {
            $scope.MachineMasterList = response.data;
        });
    }
    $scope.GetCboMachineMasterCbo();


    $scope.GetCboSkillGroupingCbo = function () {
        debugger;
        $http({
            method: 'GET',
            url: 'IE/OperationMaster/GetCboSkillGrouping'
        }).then(function successCallback(response) {
            $scope.SkillGroupingList = response.data;
        });
    }
    $scope.GetCboSkillGroupingCbo();







    $scope.GetCbolegalDesignation = function () {
        debugger;
        $http({
            method: 'GET',
            url: 'IE/OperationMaster/GetCbolegalDesignation'
        }).then(function successCallback(response) {
            $scope.legalDesignationList = response.data;
        });
    }
    $scope.GetCbolegalDesignation();


    $scope.GetCboProcess = function () {
        debugger;
        $http({
            method: 'GET',
            url: 'IE/OperationMaster/GetCboProcess'
        }).then(function successCallback(response) {
            $scope.ProcessList = response.data;
        });
    }
    $scope.GetCboProcess();
    
   
   
//#endregion


    // #region For AutoSequenceNo
    $scope.GeneratSequenceNo = function () {
        $http({
            method: 'GET',
            url: 'IE/OperationMaster/GetAutoSequence'
        }).then(function successCallback(response) {
           $scope.modelNew.Sequence = response.data;
        });
    }
    $scope.GeneratSequenceNo();

    
 //#endregion AutoSequenceNo

    // #region For AutoSequenceNo For ManPower
    $scope.GetAutoSequenceForManPower = function () {
        debugger;
        $http({
            method: 'GET',
            url: 'IE/OperationMaster/GetAutoSequenceForManPower'
        }).then(function successCallback(response) {
            $scope.modelNewM.Sequence = response.data;
        });
    }
    $scope.GetAutoSequenceForManPower();


 //#endregion AutoSequenceNo

    

 // #region Data Save Update and Delete
   

    $scope.Save = function () {
        debugger;
        angular.copy($scope.modelNew, $scope.model);
        $scope.$broadcast('show-errors-check-validity');
        try {
            if ($scope.modelNewForm.$valid) {
                if ($scope.Action === 'Save') {
                    $http({
                        method: 'POST',
                        url: $scope.saveUrl,
                        data: $scope.model,
                        dataType: 'JSON'
                    }).then(function successCallback(response) {
                        if (response.data.Error === true) {
                            //$scope.getData();
                            ShowResult(response.data.Message, 'failure');
                            throw response.data.Message;
                        }
                        else {
                            ShowResult(response.data.Message, 'success');
                            $scope.Action = 'Update';                    
                           
                            $scope.getaldataOperationMaster();
                            $scope.Clear();
                            $scope.modelNew.OperationMasterIdID = response.data.Id;  
                     
                        }
                    }), function errorCallBack(response) {
                        ShowResult(response.data.Message, 'failure');

                    };
                }
                else if ($scope.Action === 'Update') {
                    $http({
                        method: 'POST',
                        url: $scope.updateUrl,
                        data: $scope.model,
                        dataType: 'JSON'
                    }).then(function successCallback(response) {
                        if (response.data.Error === true) {
                            ShowResult(response.data.Message, 'failure');
                            //$scope.getData();
                            throw response.data.Message;
                        }
                        else {
                            ShowResult(response.data.Message, 'success');
                            $scope.getaldataOperationMaster();
                        }
                    }, function errorCallBack(response) {
                        //$scope.getData();
                        //ShowResult(response.data.Message, 'failure');
                        throw response.data.Message;
                    });
                }
           }
        } catch (e) {
            ShowResult(e, 'failure');
        }
    };

    


    $scope.Delete = function () {

        if (!baseService.isUndefinedOrNull($scope.modelNew.Id)) {
            $http({
                method: 'POST',
                url: $scope.deleteUrl + $scope.modelNew.Id,
                dataType: 'JSON'
            }).then(function (response) {
                if (response.data.Error === true)
                    ShowResult(response.data.Message, 'failure');
                else {
                    ShowResult(response.data.Message, 'success');
                    $scope.getaldataOperationMaster();
                    //ClearFields();
                }
                function errorCallBack(response) {
                    ShowResult(response.data.Message, 'failure');
                }
            });
        }
    
        else
            ShowResult('First delete all line item.', 'failure');
    };
    $scope.DeleteManpower = function () {

        if (!baseService.isUndefinedOrNull($scope.modelNewM.Id)) {
            $http({
                method: 'POST',
                url: $scope.deleteUrl1 + $scope.modelNewM.Id,
                dataType: 'JSON'
            }).then(function (response) {
                if (response.data.Error === true)
                    ShowResult(response.data.Message, 'failure');
                else {
                    ShowResult(response.data.Message, 'success');
                    $scope.getaldataOperationMaster();
                    ClearFieldss();
                }
                function errorCallBack(response) {
                    ShowResult(response.data.Message, 'failure');
                }
            });
        }

        else
            ShowResult('First delete all line item.', 'failure');
    };
    $scope.Clear = function () {
        ClearFields($scope.GeneratSequenceNo());
        return true;
    };
    $scope.Clear1 = function () {
        ClearFieldss($scope.GetAutoSequenceForManPower());
        return true;
    };
    function ClearFields(seq) {
        $scope.Action = 'Save';
        $scope.OperationMaster = {};
        $scope.modelNew = { Active: true };
        $scope.modelNew.Active = true;
        $scope.modelNew.Sequence = seq;
    }
    function ClearFieldss(seq) {
        $scope.Action1 = 'Save';
        $scope.OperationMaster = {};
        $scope.modelNewM = { Active: true };
        //$scope.modelNew.Active = true;
        $scope.modelNewM.Sequence = seq;
    }
 
 //#endregion 


    $scope.recorddoubleclick = function ($event) {
        debugger;       
        var x = $event;
        $scope.OMId = x.data.Id;
       
       // $scope.modelNew.OperationMasterIdID = response.data.Id;  
        $scope.GetDataByMasterOrderIdfn($scope.OMId);
       // $scope.GetDataByMasterOrderIdfnMP($scope.OMId);
        $scope.Action = 'Update';
       
        if (!$rootScope.isCollapsed) $rootScope.toggle();
    };
    $scope.recorddoubleclickMP = function ($event) {
        debugger;
        var x = $event;
        $scope.OMId = x.data.Id;
        $scope.OperationMasterId = x.data.OperationMasterId;
        $scope.GetDataByMasterOrderIdfnMP($scope.OMId);
        $scope.Action1 = 'Update';
        if (!$rootScope.isCollapsed) $rootScope.toggle();
    };
    $scope.GetDataByMasterOrderIdfn = function (OMId) {
        debugger;
        $http({
            method: 'GET',
            url: 'IE/OperationMaster/GetDataByMasterOrderId?id=' + OMId
        }).then(function successCallback(response) {
          
            $scope.modelNew = response.data[0];
            $scope.modelNew.OperationMasterIdID = response.data[0].Id;

        });
    }

    $scope.GetDataByMasterOrderIdfnMP = function (OMId) {
        debugger;
        $http({
            method: 'GET',
            url: 'IE/OperationMaster/GetDataByMasterOrderIdMP?id=' + OMId
        }).then(function successCallback(response) {
            $scope.modelNewM = response.data[0];


        });
    }
    $scope.tab = 1;
    $scope.setTab = function (newTab) {
        $scope.tab = newTab;
        $scope.getalldata1();

    };
    $scope.isSet = function (tabNum) {
        return $scope.tab === tabNum;
    };
    $scope.SaveManpower = function () {
        debugger;
        angular.copy($scope.modelNewM, $scope.modelM);
        $scope.modelM.OperationMasterId = $scope.modelNew.OperationMasterIdID;   
        
        $scope.$broadcast('show-errors-check-validity');
        try {
            if ($scope.modelNewForm1.$valid) {
                if ($scope.Action1 === 'Save') {
                    if ($scope.modelM.PositionId === null) {
                        ShowResult('Please select Position');
                    }
                    else if ($scope.modelM.Caption === null) {
                        ShowResult('Please input Caption');
                    }
                    else if ($scope.modelM.ManpowerBudget === null) {
                        ShowResult('Please input Manpower Budget');
                    }
                    $http({
                        method: 'POST',
                        url: $scope.saveUrl1,
                        data: $scope.modelM,
                        dataType: 'JSON'
                    }).then(function successCallback(response) {
                        if (response.data.Error === true) {
                            //$scope.getData();
                            ShowResult(response.data.Message, 'failure');
                            throw response.data.Message;
                        }
                        else {
                            ShowResult(response.data.Message, 'success');
                            $scope.Action1 = 'Update';
                            $scope.GetOperationPositionMPBudget();
                            $scope.Clear();

                        }
                    }), function errorCallBack(response) {
                        ShowResult(response.data.Message, 'failure');

                    };
                }
                else if ($scope.Action1 === 'Update') {
                    $scope.modelM.OperationMasterId = $scope.OperationMasterId;
                    if ($scope.modelM.PositionId === null) {
                        ShowResult('Please select Position');
                    }
                    else if ($scope.modelM.Caption === null) {
                        ShowResult('Please input Caption');
                    }
                    else if ($scope.modelM.ManpowerBudget === null) {
                        ShowResult('Please input Manpower Budget');
                    }
                    $http({
                        method: 'POST',
                        url: $scope.updateUrl1,
                        data: $scope.modelM,
                        dataType: 'JSON'
                    }).then(function successCallback(response) {
                        if (response.data.Error === true) {
                            ShowResult(response.data.Message, 'failure');
                            //$scope.getData();
                            throw response.data.Message;
                        }
                        else {
                            ShowResult(response.data.Message, 'success');
                            $scope.GetOperationPositionMPBudget();
                        }
                    }, function errorCallBack(response) {
                        //$scope.getData();
                        //ShowResult(response.data.Message, 'failure');
                        throw response.data.Message;
                    });
                }
            }
        } catch (e) {
            ShowResult(e, 'failure');
        }
    };
}

File: APLOS/Areas/Machines/Controllers/MachineMasterControllers.cs
using Library.Model.IE;
using Aplos.Properties;
using Library.Data;
using Library.Service.IEnumerable;
using Library.Service.Machines;
using Library.Core;
using System;
using System.Collections.Generic;
using System.IO;
using System.Web;
using System.Web.Mvc;
using System.Web.Script.Serialization;
using Library.Service.Helpers;
using System.Threading;
using Library.Crosscutting.Security;
using Library.Service.IE;
using Library.Model.Inventory;
using Library.Service.Systems;
using Library.Service.Enums;

namespace Aplos.Areas.IE.Controllers
{
    public class MachineMasterControllers : Controller
    {
        #region Constructor



        private readonly IOperationMasterService _operationMasterService;
        private readonly IOperationService _operationService;
        private readonly IOperationVariationService _operationStepService;

        private readonly IOperationTimeCaptureMasterService _ioperationtimecaptureservice;
        private readonly IOperationTimeCaptureDetailService _operationtimecapturedetailservice;
        private readonly IOperationPositionMPBudgetService _OperationPositionMPBudgetService; 

        

        public MachineMasterControllers(
            IOperationMasterService operationMasterService
            , IOperationTimeCaptureMasterService operationTimeCaptureService
            , IOperationTimeCaptureDetailService operationtimecapturedetailservice
            , IOperationService operationService
            , IOperationVariationService operationStepService
            , IOperationPositionMPBudgetService OperationPositionMPBudgetService

            )
        {
            _operationStepService = operationStepService;
            _operationtimecapturedetailservice = operationtimecapturedetailservice;
            _operationService = operationService;
            _ioperationtimecaptureservice = operationTimeCaptureService;
            _operationMasterService = operationMasterService;
            _OperationPositionMPBudgetService = OperationPositionMPBudgetService;
        }

        #endregion Constructor

        #region -- Pages

        [Authorize, HttpGet]
        public ActionResult Aplos()
        {
            return View();
        }

        #endregion -- Pages

        #region -- Operations for OperationMaster

        [Authorize, HttpGet]
        public JsonResult GetCboOperationActivity()
        {
            var identity = (CustomIdentity)Thread.CurrentPrincipal.Identity;
            return Json(_operationMasterService.GetCboCompanyGroup(), JsonRequestBehavior.AllowGet);
        }
        [Authorize, HttpGet]
        public JsonResult GetCboOperationType()
        {
            var identity = (CustomIdentity)Thread.CurrentPrincipal.Identity;
            return Json(_operationMasterService.GetCboOperationType(), JsonRequestBehavior.AllowGet);
        }


        public JsonResult GetCboOperationCategory()
        {
            var identity = (CustomIdentity)Thread.CurrentPrincipal.Identity;
            return Json(_operationMasterService.GetCboOperationCategory(), JsonRequestBehavior.AllowGet);
        }

        public JsonResult GetCboSkill()
        {
            var identity = (CustomIdentity)Thread.CurrentPrincipal.Identity;
            return Json(_operationMasterService.GetCboSkill(), JsonRequestBehavior.AllowGet);
        }


        public JsonResult GetCboMachineMaster()
        {
            var identity = (CustomIdentity)Thread.CurrentPrincipal.Identity;
            return Json(_operationMasterService.GetCboMachineMaster(), JsonRequestBehavior.AllowGet);
        }



        public JsonResult GetCboSkillGrouping()
        {
            var identity = (CustomIdentity)Thread.CurrentPrincipal.Identity;
            return Json(_operationMasterService.GetCboSkillGrouping(), JsonRequestBehavior.AllowGet);
        }

        public JsonResult GetCbolegalDesignation()
        {
            var identity = (CustomIdentity)Thread.CurrentPrincipal.Identity;
            return Json(_operationMasterService.GetCbolegalDesignation(), JsonRequestBehavior.AllowGet);
        }
        public JsonResult GetCboProcess()
        {
            var identity = (CustomIdentity)Thread.CurrentPrincipal.Identity;
            return Json(_operationMasterService.GetCboProcess(), JsonRequestBehavior.AllowGet);
        }

        public JsonResult GetCboEntity() 
        {
            var identity = (CustomIdentity)Thread.CurrentPrincipal.Identity;
            return Json(_OperationPositionMPBudgetService.GetCboEntity(), JsonRequestBehavior.AllowGet);
        }
        public JsonResult GetCboPosition()
        {
            var identity = (CustomIdentity)Thread.CurrentPrincipal.Identity;
            return Json(_OperationPositionMPBudgetService.GetCboPosition(), JsonRequestBehavior.AllowGet);
        }




        #endregion


        #region Grid data for Operation Master UI
        // [Authorize, HttpGet]
        public JsonResult GetOperationMaster()
        {
            var identity = (CustomIdentity)Thread.CurrentPrincipal.Identity;
            return Json(_operationMasterService.GetOperationMaster(), JsonRequestBehavior.AllowGet);
        }
       // [Authorize, HttpGet]
        public JsonResult GetOperationPositionMPBudget()
        {
            var identity = (CustomIdentity)Thread.CurrentPrincipal.Identity;
            return Json(_OperationPositionMPBudgetService.GetOperationPositionMPBudgetService(), JsonRequestBehavior.AllowGet);
        }

        #endregion


        #region GetData by Operation Master Id
        //[Authorize, HttpGet]
        public JsonResult GetDataByMasterOrderId(string id)
        {
            var identity = (CustomIdentity)Thread.CurrentPrincipal.Identity;
            return Json(_operationMasterService.GetDataByMasterOrderId(id), JsonRequestBehavior.AllowGet);
        }
        [Authorize, HttpGet]
        public JsonResult GetDataByMasterOrderIdMP(string id) 
        {
            var identity = (CustomIdentity)Thread.CurrentPrincipal.Identity;
            return Json(_OperationPositionMPBudgetService.GetDataByMasterOrderIdMP(id), JsonRequestBehavior.AllowGet);
        }

        #endregion

        [HttpGet, Authorize]
        public JsonResult GetAutoSequence()
        {
            return Json(_operationMasterService.GetAutoSequence(), JsonRequestBehavior.AllowGet);
        }
        [HttpGet, Authorize]
        public JsonResult GetAutoSequenceForManPower()
        {
            return Json(_OperationPositionMPBudgetService.GetAutoSequence(), JsonRequestBehavior.AllowGet);
        }
        private string GetPK()
        {
            var identity = (CustomIdentity)Thread.CurrentPrincipal.Identity;
            return _operationMasterService.GetAutoNumber(nameof(OperationMaster), PKGeneratorEnum.Yearly, identity.CompanyGroupId, DateTime.Now);
        }
        [HttpPost]
        public JsonResult Create(OperationMaster model)
        {
            try
            {
                _operationMasterService.Check(model);
            }
            catch(CustomException)
            {
                throw;
            }
          
            var identity = (CustomIdentity)Thread.CurrentPrincipal.Identity;
            model.Id = "OP" + GetPK();
            model.CompanyGroupID = identity.CompanyGroupId;
            if(model.Type=="ACTIVITY")
            {
                model.MachineMasterId = null;
                model.Skillid = model.Skillid;
            }
            else if (model.Type == "OPERATION")
            {
                model.MachineMasterId = model.MachineMasterId;
                model.Skillid = null;
            }
            else if (model.Active)
            {
                model.Active=true;
            }
            else if (!model.Active)
            {
                model.Active = false;
            }

            _operationMasterService.Insert(model);
            return Json(new { OperationMaster = model,model.Id, Sequence = _operationMasterService.GetAutoSequence(), Message = AplosMessage.Insert });
        }
        [HttpPost]
        public JsonResult CreateManpower(OperationPositionMPBudget model)
        {
            try
            {
                _OperationPositionMPBudgetService.Check(model);
            }
            catch (CustomException)
            {
                throw;
            }

            var identity = (CustomIdentity)Thread.CurrentPrincipal.Identity;
            model.Id = "OPP" + GetPK();
            model.CompanyGroupID = identity.CompanyGroupId;            
           if (model.Active)
            {
                model.Active = true;
            }
            else if (!model.Active)
            {
                model.Active = false;
            }
            else if (model.PositionId==null)
            {
                throw new CustomException("Please select position");
            }
            
            else if (model.PositionId == null)
            {
                throw new CustomException("Please select position");
            }
            else if (model.Caption == null)
            {
                throw new CustomException("Please input Caption");
            }
            else if (model.ManpowerBudget == '0')
            {
                throw new CustomException("Please input Manpower Budget");
            }
            _OperationPositionMPBudgetService.Insert(model);
            return Json(new { OperationMaster = model, Sequence = _operationMasterService.GetAutoSequence(), Message = AplosMessage.Insert });
        }
        [HttpPost]
        public JsonResult Edit(OperationMaster model)
        {
            try
            {
                _operationMasterService.Check(model);
            }
            catch (CustomException)
            {
                throw;
            }
            if (model.Type == "ACTIVITY")
            {
                model.MachineMasterId = null;
            }
            else if (model.Type == "OPERATION")
            {
                model.MachineMasterId = model.MachineMasterId;
            }
            else if (model.Active)
            {
                model.Active = true;
            }
            else if (!model.Active)
            {
                model.Active = false;
            }
            _operationMasterService.Update(model);
            return Json(new { Sequence = _operationMasterService.GetAutoSequence(), Message = AplosMessage.Updated });
        }
        [HttpPost]
        public JsonResult EditManpower(OperationPositionMPBudget model)
        {
            try
            {
                _OperationPositionMPBudgetService.Check(model);
            }
            catch (CustomException)
            {
                throw;
            }            
           if (model.Active)
            {
                model.Active = true;
            }
            else if (!model.Active)
            {
                model.Active = false;
            }
            else if (model.PositionId == null)
            {
                throw new CustomException("Please select position");
            }
            else if (model.Caption == null)
            {
                throw new CustomException("Please input Caption");
            }
            else if (model.ManpowerBudget == '0')
            {
                throw new CustomException("Please input Manpower Budget");
            }
            _OperationPositionMPBudgetService.Update(model);
            return Json(new { Sequence = _operationMasterService.GetAutoSequence(), Message = AplosMessage.Updated });
        }
        [HttpPost]
        public ActionResult Delete(string id)
        {
            _operationMasterService.Delete(id);
            return Json(new { Sequence = _operationMasterService.GetAutoSequence(), Message = AplosMessage.Deleted });
        }
        [HttpPost]
        public ActionResult DeleteManpower(string id)
        {
            _OperationPositionMPBudgetService.Delete(id);
            return Json(new { Sequence = _operationMasterService.GetAutoSequence(), Message = AplosMessage.Deleted });
        }

    }

}